### PR TITLE
Some internal changes

### PR DESCRIPTION
**Pull request by archermarx**
This PR rearranges the code internals for very slightly better performance (1-2%), but more importantly better separates the electron part of the solver loop from the ion part of the solver loop. This architectural change will help facilitate later changes on the ion side without needing to mess with the electron part at all. 

Most notably, this PR makes it so now, electron functions do not use the heavy species state matrix `U`. This means that the details about how this is laid out are left to the ions. The electrons now just require that `ne`, `ni`, `niui`, `ji`, and `avg_Z` in the `cache` struct are set.

Additionally, I have made some changes to the `Config` struct
- Removed TransitionFunction and associated `transition_function` option in `Config`. Replaced with `transition_length` which can be specified either as a `Float64` in meters or as a Unitful length. The reason behind this change is that nobody uses the alternate transition functions, and it lets us remove one specialization on the `Config` struct, potentially reducing compile times. 
- Removed the `solve_background_neutrals` option. This option is redundant, and setting the back-pressure to zero (the default) accomplishes the same thing

Finally, I removed the infrastructure for multiple neutral fluids. This was vestigial anyway, and I have better ideas for how we might want to handle this in the future.

### CODE REVIEW

File: src/simulation/update_electrons.jl [hunk 1]
```diff
@@ -1,67 +1,20 @@
...
     (;
         B, ue, Tev, ∇ϕ, ϕ, pe, ne, nϵ, μ, ∇pe, νan, νc, νen, νei, radial_loss_frequency,
-        Z_eff, νiz, νex, νe, ji, Id, νew_momentum, κ, ni, ui, Vs, nn, nn_tot, niui,
+        Z_eff, νiz, νex, νe, ji, Id, νew_momentum, κ, Vs, nn, ϵ_tot, K,
         Id_smoothed, smoothing_time_constant, anom_multiplier,
         errors, channel_area
     ) = params.cache
 
-    # Allow for system interrupts
-    yield()
```
Comment (DecBrick) [inline src/simulation/update_electrons.jl:12]: Does the removal of the yield statement still allow for interrupts?

Comment (archermarx) [reply src/simulation/update_electrons.jl:12]: It's now in `solution.jl`

**Comment by DecBrick** [comment]
I think generally these changes are a trend in the right direction. My main concern is that removing the multiple neutral infrastructure will make it harder to properly implement charge exchange in the future. However, if we are not terrible concerned about this functionality, I think the performance and reability improvement is worth it

**Comment by archermarx** [comment]
@DecBrick that's a good point, but I think there are better ways we can implement it. I am leaning toward a future version where we treat the neutrals kinetically (either like Hall2De or using particles)

**Review by DecBrick — APPROVED** [review]

**Pull request closed without merging**